samcli/cli/types.py

```diff
@@ -222,13 +248,36 @@ def convert(self, value, param, ctx):
                 for k in tags:
                     self._add_value(result, _unquote_wrapped_quotes(k), _unquote_wrapped_quotes(tags[k]))
             else:
-                groups = re.findall(self._pattern, val)
-
-                if not groups:
-                    fail = True
-                for group in groups:
-                    key, v = group
-                    self._add_value(result, _unquote_wrapped_quotes(key), _unquote_wrapped_quotes(v))
+                # Instead of parsing a full {tag}={tag} pattern, we will try to look for quoted string with spaces
```

Comment:
Can we refactor this new logic out to a separate method to unit test? There's a lot of moving parts here

Reply:
This is tricky because the block below

```
groups = re.findall(self._pattern, val)
```

take longer time to return